clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -845,42 +844,93 @@ raw2trace_t::process_header(raw2trace_thread_data_t *tdata)
 }
 
 #ifdef BUILD_PT_POST_PROCESSOR
+
+std::unique_ptr<pt_data_buf_t>
+raw2trace_t::get_next_kernel_entry(raw2trace_thread_data_t *tdata,
+                                   std::unique_ptr<pt_metadata_buf_t> &pt_metadata,
+                                   uint64_t expected_syscall_idx)
+{
+    DR_ASSERT(tdata->kthread_file != nullptr);
+    if (tdata->kthread_file->eof())
+        return nullptr;
+    if (!tdata->pt_metadata_processed) {
+        log(2, "Reading PT metadata for tid " INT64_FORMAT_STRING "\n", tdata->tid);
+        pt_metadata = std::unique_ptr<pt_metadata_buf_t>(new pt_metadata_buf_t());
+        if (!tdata->kthread_file->read((char *)&pt_metadata->header[0],
```

Comment:
Prefer C++-style casts.  Ditto for the 2 similar ones below.

Reply:
Done.